gateway/ln-gateway/src/lib.rs

```diff
@@ -542,7 +542,18 @@ impl Gateway {
         if handle
             .cancel_on_shutdown(async move {
                 loop {
-                    let state = self.get_state().await;
+                    let payment_request = tokio::select! {
+                        payment_request = stream.next() => {
+                            payment_request
+                        }
+                        () = self.is_shutting_down_safely() => {
+                            break;
+                        }
+                    };
+
+                    // Hold the Gateway state's lock so that it doesn't change before `handle_htlc`.
+                    let state_guard = self.state.read().await;
+                    let state = state_guard.clone();
```

Comment:
That does not compile, you cannot deference the `state_guard` because that would move it, and it doesn't implement `Copy`. This is why I am cloning it, so that we can actually read the state.

Reply:
Did you try it? It runs on my machine. Using `ref lightning_context` and `*state_guard` performs a dereference rather than a clone

Reply:
ah I missed the `ref` in this. Yes this works, I will fix.